test/run/run-e2e-tests.js

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env node
+
+const { promisify } = require( 'util' );
+const { initApp } = require( './init_app' );
+const { initLogs } = require( './init_logs' );
+const { initTests } = require( './init_tests' );
+const { initDriver } = require( './init_driver' );
+const { checkCredentials } = require( './checks' );
+
+let app;
+let driver;
+
+const delay = promisify( setTimeout );
+
+// Handle both user-initiated (SIGINT) and normal termination.
+process.on( 'SIGINT', function() {
+    handleExit();
+    process.exit();
+} );
+
+process.on( 'exit', handleExit );
+function handleExit() {
+    if ( driver ) {
+        driver.kill();
+    }
+    if ( app ) {
+        app.kill();
+    }
+}
+
+async function run() {
+    try {
+        checkCredentials();
```

Comment:
I'l make a similar comment here about my own anxiety and obscuring what's happening vs. hiding _how_ it's happening. this doesn't give me much information about the credentials even though I assume it's checking _something_.

Another formulation might abstract the check and expose the things being checked…

```js
async function run() {
  const requriedENVs = [ 'E2EUSERNAME', 'E2EPASSWORD', 'E2E_MAILOSAUR_INBOX' ];
  const missingENVs = requiredENVs.filter( name => ! process.env[name] || process.env[name] === '' );
  if ( missingENVs.length ) {
    throw new Error( `Missing non-empty ENV for: ${ missingENVs.join(', ')` );
  }

  …
}
```

Reply:
Agreed - see my response to your prior comment. Everything's been refactored inline. Also, I like the above as a much more concise way to check the required environment variables. See refactor [here](https://github.com/Automattic/wp-desktop/pull/778/files#diff-95f02e568cfd6b3a0b1830f8fd2306a1R59).